### PR TITLE
Api: Add support for bucket notifications

### PR DESCRIPTION
**Pull request by poornas**
Add support for GetBucketNotification, SetBucketNotification and RemoveAllBucketNotifications

### CODE REVIEW

File: Minio/DataModel/Notification/BucketNotification.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        public bool ShouldSerializeName()
+        {
+            return this.Name != null;
+        }
+        // Serializes the notification configuration as an XML string
+        public string ToXML()
+        {
+            XmlWriterSettings settings = new XmlWriterSettings();
+            settings.OmitXmlDeclaration = true;
+            MemoryStream ms = new MemoryStream();
```
Comment (krisis) [inline Minio/DataModel/Notification/BucketNotification.cs]: Does it make sense to use `using` block to make `MemoryStream` finalization automagic even in the event of exceptions?

**Comment by poornas** [comment]
@deekoder - apaprently windows 8 is compatible with VS2017 (https://www.visualstudio.com/en-us/productinfo/vs2017-compatibility-vs). YMMV

**Comment by deekoder** [comment]
I am installing and testing. need till EOW to confirm on these changes. sorry about the delay.

File: Minio/DataModel/Notification/BucketNotification.cs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+        public BucketNotification()
+        {
+            LambdaConfigs = new List<LambdaConfig>();
+            TopicConfigs = new List<TopicConfig>();
+            QueueConfigs = new List<QueueConfig>();
+        }
+        // AddTopic adds a given topic config to the general bucket notification config
+        public void AddTopic(TopicConfig topicConfig)
+        {
```
Comment (poornas) [inline Minio/DataModel/Notification/BucketNotification.cs]: fixed

File: Minio/DataModel/Notification/BucketNotification.cs (hunk 1 above)
Comment (poornas) [inline Minio/DataModel/Notification/BucketNotification.cs]: that's a good idea.done

**Comment by harshavardhana** [comment]
> I am installing and testing. need till EOW to confirm on these changes. sorry about the delay.

@deekoder @krisis @poornas what else is pending in this PR?

**Comment by poornas** [comment]
ping @deekoder

**Comment by deekoder** [comment]
Followed instructions to setup testing. Still need more info to complete review.

**Comment by krisis** [comment]
@deekoder what kind of testing are you doing for this PR? I need a day or two to set Visual Studio on a windows laptop and run unit tests and the bucket notifications examples.

**Comment by harshavardhana** [comment]
I suggest we take this in. Do you still want to test and valdiate this @deekoder ?

**Comment by deekoder** [comment]
I could not validate it. It was too cumbersome to set it up and run. Ok lets go ahead and take it.

**Pull request merged**